src/lib/layout/Footer.svelte

```diff
@@ -0,0 +1,144 @@
+<script lang="ts">
+  import { onMount } from 'svelte';
+  import Icon from '$lib/component/Icon.svelte';
+  import Icons from '$lib/component/Icons';
+
+  function copyToClipboard(content: string) {
+    navigator.clipboard.writeText(content);
+  }
+
+  onMount(() => {
+    const iconsContainer: NodeListOf<Element> | null = document.querySelectorAll('.footer-icons');
+    if (iconsContainer !== null) {
+      iconsContainer.forEach((container) => {
+        const icons: HTMLCollection = container.children;
+
+        for (let i of icons) {
+          i.addEventListener('mouseover', (e) => {
+            const el = e.target as HTMLElement;
+            if (el !== null) {
+              el.style.opacity = '1.0';
+              for (let j of icons) {
+                if (j !== el) {
+                  (j as HTMLElement).style.opacity = '0.5';
+                }
+              }
+            }
+          });
+
+          i.addEventListener('mouseleave', (e) => {
+            const el = e.target as HTMLElement;
+            if (el !== null) {
+              el.style.opacity = '1.0';
+              for (let j of icons) {
+                if (j !== el) {
+                  (j as HTMLElement).style.opacity = '1.0';
+                }
+              }
+            }
+          });
+        }
+      });
+    }
+  });
+</script>
+
+<div class="z-10 w-full bg-transparent p-3 max-sm:hidden">
```

Comment:
I'm not sure what's the best or most common way. But what do you think about separating mobile/desktop components to reduce the file size? We could still have a single `Footer` to abstract them and use the onMount. I'd like to hear your opinion, I think this is a matter of preference, there is no better way

Reply:
I personally prefer the current implementation as it's what I'm most familiar with. However, if there are substantial changes between the screen sizes versions of a component the suggested method may be useful, but as for this specific case I do not think it would be that beneficial.